lldb/test/Shell/SymbolFile/cas-gmodule.test

```diff
@@ -0,0 +1,46 @@
+# REQUIRES: system-darwin
+# RUN: rm -rf %t
+# RUN: split-file %s %t
+# RUN: sed "s|DIR|%/t|g" %t/cas-config.template > %t/.cas-config
+
+# RUN: %python %S/../../../scripts/clang-explicit-module-build.py -cas-path %t/cas -- %clang_host -c -g -o %t/test.o %t/tu.c -fmodules -gmodules -fmodules-cache-path=%t/module-cache
+# RUN: %clang_host %t/test.o -o %t/main
+# RUN: %lldb %t/main -s %t/lldb.script 2>&1 | FileCheck %s
+# CHECK: loading module 'Bottom' using CASID
+# CHECK: loading module 'Top' using CASID
+# CHECK: top = (x = 1)
```

Comment:
Like in the dsymutil patch, it would be important to also test at least the most common error paths (config file doesn't exist, build dir has been removed, CAS is using different file storage format,...)